crates/db_schema/src/source/modlog.rs

```diff
@@ -0,0 +1,62 @@
+use crate::newtypes::{CommentId, CommunityId, InstanceId, ModlogId, PersonId, PostId};
+use chrono::{DateTime, Utc};
+#[cfg(feature = "full")]
+use i_love_jesus::CursorKeysModule;
+use lemmy_db_schema_file::enums::ModlogKind;
+#[cfg(feature = "full")]
+use lemmy_db_schema_file::schema::modlog;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+#[cfg_attr(
+  feature = "full",
+  derive(Queryable, Selectable, Identifiable, CursorKeysModule)
+)]
+#[cfg_attr(feature = "full", diesel(table_name = modlog))]
+#[cfg_attr(feature = "full", diesel(check_for_backend(diesel::pg::Pg)))]
+#[cfg_attr(feature = "ts-rs", derive(ts_rs::TS))]
+#[cfg_attr(feature = "ts-rs", ts(optional_fields, export))]
+#[cfg_attr(feature = "full", cursor_keys_module(name = modlog_keys))]
+pub struct Modlog {
+  pub id: ModlogId,
+  pub kind: ModlogKind,
+  pub removed: bool,
+  pub mod_id: PersonId,
+  pub reason: Option<String>,
+  #[serde(skip)]
+  pub target_person_id: Option<PersonId>,
+  #[serde(skip)]
+  pub target_community_id: Option<CommunityId>,
+  #[serde(skip)]
+  pub target_post_id: Option<PostId>,
+  #[serde(skip)]
+  pub target_comment_id: Option<CommentId>,
+  #[serde(skip)]
+  pub target_instance_id: Option<InstanceId>,
+  #[serde(skip)]
+  pub expires_at: Option<DateTime<Utc>>,
+  pub published_at: DateTime<Utc>,
+}
+
+#[derive(derive_new::new)]
+#[cfg_attr(feature = "full", derive(Insertable))]
+#[cfg_attr(feature = "full", diesel(table_name = modlog))]
+pub struct ModlogInsertForm {
```

Comment:
This is one place where I do think we should be strict about the different types of inserts, and get rid of this generic insert form. 

We should create specific and strict `Insertable` forms for each of the types: https://github.com/diesel-rs/diesel/blob/2.3.x/examples/postgres/all_about_inserts/src/lib.rs#L23-L30

I know its verbose, but it'll prevent us from inserting incorrect modlog data.